### ISSUE TITLE
validate string instead of a file

### ISSUE THREAD

**Issue by mrothenbuecher** [jupitex/node-libxml#1]
Hi,

is it possible to validate a string which contains xml instead of a xml-file. (same for dtd-Files)?

Will it be a feature in the future?

**Comment by MatthD** [comment]
Hi!
I think this could be easily implemented, I will have to find the good libxml library function

**Comment by MatthD** [comment]
OK I release the 3.2.0 update.
https://github.com/jupitex/node-libxml/releases/tag/3.2.0
Available on npm too, now you can load XML File.

You can load xml via the _libxml.loadXmlFromString(_) function

I started to work on load dtd but I have a problem... node-libxml has been thought to be able to validate against multiple dtd/xsd, to do this We return the path which validate because there is no identifier in the dtd string.
I need to find a way to return the dtd/xsd which has validated

**Comment by MatthD** [comment]
Yeah but how people will do the link between the hash & the variable they sent ?
For eg:

your can do :  

```
...
    let xml = fs.require("myxml.xml");
    libxml.loadDtds(['dtd1.path','dtd2.path']);
    let resultsDTD =  libxml.validateAgainstDtds();
...
```javascript

Will currently return the path of the which one has validate & if no one validate will return all the errors.
How do you image the use with string utilisation ?

**Comment by onsightit** [comment]
Whatever change you just made broke my application.  What changes need to be made to update my application?

**Comment by MatthD** [comment]
Hum I am sorry to ear that @onsightit . what is the error you got ?

What is your os, which function do you use ?

be sure to use V3.2.3

**Comment by onsightit** [comment]
It's the pre-built binares.  I have a node server that does not have g++ installed.

**Comment by onsightit** [comment]
Weird, the tar.gz is downloadable, but the download fails using 'npm update'.

**Comment by onsightit** [comment]
I had to go back to 3.1.0 to get the install to work.
